recipes/winpcap/all/conanfile.py

```diff
@@ -0,0 +1,134 @@
+
+required_conan_version = ">=1.33.0"
+
+
+class WinPcapConan(ConanFile):
+    name = "winpcap"
+    description = "For many years, WinPcap has been recognized as the industry-standard tool for link-layer network access in Windows environments, " \
+                  "allowing applications to capture and transmit network packets bypassing the protocol stack, and including kernel-level packet filtering, " \
+                  "a network statistics engine and support for remote packet capture."
+    license = "BSD-3-Clause"
+    topics = ("conan", "packet", "capture", "filtering", "network", "protocol")
+    homepage = "https://www.winpcap.org/"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "capture_type": ["bpf", "dag", "dlpi", "enet", "nit", "null", "libdlpi", "linux", "pf", "sita", "snit", "snoop", "win32"],
+        "remote": [True, False],
+        "bluetooth": [True, False],
+        "usb_sniffing": [True, False],
+        "usb_sniffing_device": "ANY",
+        "turbocap": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "capture_type": "null",
+        "remote": True,
+        "bluetooth": False,
+        "usb_sniffing": True,
+        "usb_sniffing_device": "/dev/usbmon",
+        "turbocap": False,
+    }
+
+    exports_sources = "CMakeLists.txt", "cmake/*", "patches/*"
+    generators = "cmake"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+            self.options.turbocap = True
+            self.options.capture_type = "win32"
+        elif self.settings.os == "Linux":
+            self.options.capture_type = "linux"
+        if self.settings.os != "Linux":
+            del self.options.bluetooth
+            del self.options.usb_sniffing
+            del self.options.usb_sniffing_device
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.capture_type == "dag":
+            # FIXME: missing dag recipe
+            raise ConanInvalidConfiguration("missing dag cci recipe")
+        if self.options.get_safe("bluetooth"):
+            # FIXME: missing bluez recipe
+            raise ConanInvalidConfiguration("missing bluez cci recipe")
+        if self.options.turbocap and self.settings.os != "Windows":
+            # FIXME: missing turbocap recipe
+            raise ConanInvalidConfiguration("missing turbocap cci recipe")
+
+    def build_requirements(self):
+        if tools.os_info.is_windows:
+            self.build_requires("winflexbison/2.5.24")
+        else:
+            self.build_requires("bison/3.7.1")
+            self.build_requires("flex/2.6.4")
```

Comment:
here is the issue with windows.

Reply:
But it's a build requirement. I thought those didn't influence the package id?

Reply:
maybe test:

`if (hasattr(self, "settings_build") and self.settings_build.os == "Windows) or (not hasattr(self, "settings_build") and self.settings.os == "Windows")`

Reply:
Thanks! Let's try the `settings_build` approach.

Reply:
> But it's a build requirement. I thought those didn't influence the package id?

Not related to package id. In CI of CCI, a linux machine runs some tests first, and `tools.os_info.is_windows` is always False. @jgsogo can explain better than me.

Reply:
Yes. In CCI we first compute the packageIDs for all the configurations (this computation runs on a Linux machine) so we can remove duplicates and those that are invalid-config. Then, we will use win/linux/mac nodes to build the libraries. 

If (running in Linux) Conan computes an invalid-config for a given configuration (a Windows one) we won't try to build that configuration, we will just report it is invalid. For this reason, it is very important that everything used in a recipe is provided as input (via profiles) and nothing is guessed... and this is mandatory for the source lines that are executed while computed the package-id in the Linux machine (tomorrow it could be a Windows node or Macos).

Although build-requires will not affect the package-id, while computing if it is a valid configuration, we also take into account the build-requires that will be required to build. If these build-requires are not available, then the configuration is also removed.

TL;DR; `toos.os_info` is evil outside _build_ step (`build()` always takes place in the build machine).

Reply:
Thanks for the reply.
I know you had to write this message a lot already. 😅

Reply:
Should this go in the FAQ?

Reply:
See https://github.com/conan-io/hooks/pull/320 for a first attempt to implement this into a hook.

Reply:
Amazing! this is supper helpful to have and know